hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/StreamSourceStageTest.java

```diff
@@ -165,6 +167,29 @@ public void when_sparseItemsWithIngestionTimestamps_then_noExtraLatency() {
         job.cancel();
     }
 
+    @Test
+    @Category(NightlyTest.class)
+    public void when_sparseItemsWithIngestionTimestamps_then_windowIsNotEmittedTooEarly() {
+        IList<WindowResult<Long>> sinkList = instance.getList(randomMapName());
+        IMap<Integer, Integer> map = instance.getMap(randomMapName());
+
+        Pipeline p = Pipeline.create();
+        p.readFrom(Sources.mapJournal(map, START_FROM_OLDEST))
+         .withIngestionTimestamps()
+         .window(WindowDefinition.session(15_000))
+         .aggregate(AggregateOperations.counting())
+         .writeTo(Sinks.list(sinkList));
+
+        Job job = instance.newJob(p);
+        assertEquals(0, sinkList.size());
+        map.put(5, 5);
+
+        assertTrueAllTheTime(() -> assertEquals(0, sinkList.size()), 10);
```

Comment:
This is time-sensitive. If it takes 5 secs to submit the job, this assert can fail.

We can instead start measuring before submitting the job and we can safely check that more than 15 seconds passed before the first event appeared.

```
long start = System.nanoTime();
newJob(p);
assertTrueEventually( /* one item in sinkList */);
assertTrue(System.nanoTime() - start > 15 seconds);
```